org.eclipse.jdt.core.tests.compiler/src/org/eclipse/jdt/core/tests/compiler/parser/DietRecoveryTest.java

```diff
@@ -6194,72 +6052,144 @@ public void _test101() {
-		";}	\n"+
-		"public class Hello{	\n"+
-		"private static int x;	\n"+
-		"private String blah;	\n"+
-		"public static void main (String[] args){	\n"+
-		"}	\n"+
-		"public void hello (){	\n"+
-		"}	\n"+
-		"public boolean blah (){	\n"+
-		"return false;}	\n"+
-		"public void foo (){	\n"+
-		"}	\n"+
-		"}	\n";
-
-	String expectedDietUnitToString =
-		"public class Hello {\n" +
-		"  private static int x;\n" +
-		"  private String blah;\n" +
-		"  public Hello() {\n" +
-		"  }\n" +
-		"  <clinit>() {\n" +
-		"  }\n" +
-		"  public static void main(String[] args) {\n" +
-		"  }\n" +
-		"  public void hello() {\n" +
-		"  }\n" +
-		"  public boolean blah() {\n" +
-		"  }\n" +
-		"  public void foo() {\n" +
-		"  }\n" +
-		"}\n" ;
-
-	String expectedDietPlusBodyUnitToString =
-		"public class Hello {\n" +
-		"  private static int x;\n" +
-		"  private String blah;\n" +
-		"  public Hello() {\n" +
-		"    super();\n" +
-		"  }\n" +
-		"  <clinit>() {\n" +
-		"  }\n" +
-		"  public static void main(String[] args) {\n" +
-		"  }\n" +
-		"  public void hello() {\n" +
-		"  }\n" +
-		"  public boolean blah() {\n" +
-		"    return false;\n" +
-		"  }\n" +
-		"  public void foo() {\n" +
-		"  }\n" +
-		"}\n";
-
-	String expectedFullUnitToString = expectedDietUnitToString;
+		"""
+		void ___eval() {\t
+		new Runnable(){\t
+		void ___run() throws Throwable {\t
+		return blah;\t
+		}\t
+		private String blarg;\t
+		public void run (){\t
+				class Local {\s
+					void baz() {\t
+					}\t
+				}\t\s
+		}\t
+		}\t
+		;}\t
+		public class Hello{\t
+		private static int x;\t
+		private String blah;\t
+		public static void main (String[] args){\t
+		}\t
+		public void hello (){\t
+		}\t
+		public boolean blah (){\t
+		return false;}\t
+		public void foo (){\t
+		}\t
+		}\t
+		""";
+
+	String expectedDietUnitToString = """
+final class <unnamed class> {
+  public class Hello {
+    private static int x;
+    private String blah;
+    <clinit>() {
+    }
+    public Hello() {
+    }
+    public static void main(String[] args) {
+    }
+    public void hello() {
+    }
+    public boolean blah() {
+    }
+    public void foo() {
+    }
+  }
+  <unnamed class>() {
+  }
+  void ___eval() {
+  }
+}
+""";
+
+	String expectedDietPlusBodyUnitToString = """
+final class <unnamed class> {
+  public class Hello {
+    private static int x;
+    private String blah;
+    <clinit>() {
+    }
+    public Hello() {
+      super();
+    }
+    public static void main(String[] args) {
+    }
+    public void hello() {
+    }
+    public boolean blah() {
+      return false;
+    }
+    public void foo() {
+    }
+  }
+  <unnamed class>() {
+    super();
+  }
+  void ___eval() {
+    new Runnable() {
+      private String blarg;
+      void ___run() throws Throwable {
+        return blah;
+      }
+      public void run() {
+        class Local {
+          Local() {
+            super();
+          }
+          void baz() {
+          }
+        }
+      }
+    };
+  }
+}
+""";
+
```

Comment:
Are these changes just changing to text block or anything more?

Reply:
The source code that's being tested has not changed, but the expected output has changed. Since the file contains methods that are not in any class body, the parser parses the code as an unnamed class, and the printed parsed class reflects this change